stl/inc/format

```diff
@@ -2557,8 +2557,221 @@ template <_Format_supported_charT _CharT, class _Traits>
+        return _Ptr[_Index];
+    }
+};
+
+struct _Fmt_buffer_traits {
+    explicit _Fmt_buffer_traits(size_t) {}
+    size_t count() const {
+        return 0;
+    }
+    size_t limit(size_t _Size) {
+        return _Size;
+    }
+};
+
+class _Fmt_fixed_buffer_traits {
+private:
+    size_t _Count = 0;
+    size_t _Limit;
+
+public:
+    explicit _Fmt_fixed_buffer_traits(size_t _L) : _Limit(_L) {}
+    size_t count() const {
+        return _Count;
+    }
+    size_t limit(size_t _Size) {
+        size_t _N = _Limit > _Count ? _Limit - _Count : 0;
+        _Count += _Size;
+        return _Size < _N ? _Size : _N;
+    }
+};
+
+template <class _OutputIt, class _T, class _Traits = _Fmt_buffer_traits>
+class _Fmt_iterator_buffer final : public _Traits, public _Fmt_buffer<_T> {
+private:
+    _OutputIt _Out;
+    enum { _Buffer_size = 256 };
+    _T _Data[_Buffer_size];
+
+protected:
+    void grow(size_t) final override {
+        if (this->size() == _Buffer_size)
+            flush();
+    }
+    void flush();
+
+public:
+    explicit _Fmt_iterator_buffer(_OutputIt _O, size_t _N = _Buffer_size)
+        : _Traits(_N), _Fmt_buffer<_T>(_Data, 0, _Buffer_size), _Out(_O) {}
+    ~_Fmt_iterator_buffer() {
+        flush();
+    }
+
+    _OutputIt out() {
+        flush();
+        return _Out;
+    }
+    size_t count() const {
+        return _Traits::count() + this->size();
+    }
+};
+
+template <class _T>
+class _Fmt_iterator_buffer<_T*, _T> final : public _Fmt_buffer<_T> {
+protected:
+    void grow(size_t) final override {}
+
+public:
+    explicit _Fmt_iterator_buffer(_T* _Out, size_t = 0) : _Fmt_buffer<_T>(_Out, 0, ~size_t()) {}
+
+    _T* out() {
+        return &*this->end();
+    }
+};
+
+template <class _Container>
+class _Fmt_iterator_buffer<_STD back_insert_iterator<_Container>,
+    enable_if_t<_STD contiguous_iterator<typename _Container::iterator>, typename _Container::value_type>>
+    final : public _Fmt_buffer<typename _Container::value_type> {
+private:
+    _Container& _Cont;
+
+    struct _Accessor : _STD back_insert_iterator<_Container> {
+        _Accessor(_STD back_insert_iterator<_Container> _Iter) : back_insert_iterator<_Container>(_Iter) {}
+        using _STD back_insert_iterator<_Container>::container;
+    };
+
+protected:
+    void grow(size_t _Capacity) final override {
+        _Cont.resize(_Capacity);
+        this->set(&_Cont[0], _Capacity);
+    }
+
+public:
+    explicit _Fmt_iterator_buffer(_Container& _C)
+        : _Fmt_buffer<typename _Container::value_type>(_C.size()), _Cont(_C) {}
+    explicit _Fmt_iterator_buffer(_STD back_insert_iterator<_Container> _Out, size_t = 0)
+        : _Fmt_iterator_buffer(*_Accessor(_Out).container) {}
+
+    _STD back_insert_iterator<_Container> out() {
+        return _STD back_inserter(_Cont);
+    }
+};
+
+template <class _T>
+class _Fmt_counting_buffer final : public _Fmt_buffer<_T> {
+private:
+    enum { _Buffer_size = 256 };

```

Comment:
Should reuse the `_Fmt_buffer_size` from above

Reply:
They are unrelated but I can replace them with a single constant.